### ISSUE TITLE
Unordered output after incremental backups

### ISSUE THREAD

**Issue by frainz-de** [kensanata/mastodon-backup#61]
When making an incremental backup a few days after the first one, the new toots are appended to the json archive. They also appear at the end of the text and html output even if the original order had been from newest to oldest.
The straight forward solution would probably to just sort by toot date but I'm unsure how well that scales when the archive is very big.

**Comment by kensanata** [comment]
I think the correct solution would be to sort them on output. If you're using the text command to search toots, there will be very few toots to sort, and if you're producing HTML output, then you have time.
As we already load the entire JSON into memory, very big archives will need to be split and where as such a command already exists, split archives are not handled very well by all the other commands so eventually something will have to be implemented.

**Comment by kensanata** [comment]
Do you want to work on that? :smiley:

**Comment by frainz-de** [comment]
I have the feeling that you'll get me there eventually :) I'll try to dive into the code in the next days but won't promise anything as I'm supposed to be doing things for uni

**Comment by kensanata** [comment]
Take a look at 3de4398, that should do it for text output, I hope?

**Comment by frainz-de** [comment]
Yes, it seems to work. Do you think it should be sorted when not reversed as well?

**Comment by kensanata** [comment]
I don’t know. I never depend on the sorting. I just search my archive for that ONE toot I remember – so I don’t really care about the order. What do you think? I guess both directions being ordered is “clean” because otherwise we have three states: ordered, reverse ordered, and semi-ordered (the current default).

**Comment by frainz-de** [comment]
Yes, I agree. Both types of output should be sorted by date for both directions.

**Comment by kensanata** [comment]
So I'll await your patch for HTML? I am trying to rope you in...

**Comment by cangyuyao** [comment]
I just updated this tool after a long time... and seems that this issue is only for the new version? Just curious why we need to append the new statuses at the end while the statuses backed up each time are still in reverse chronological order?

**Comment by kensanata** [comment]
I think that’s just the way things are: we append new statuses at the end, and we get them in reverse chronological order. What the issue is about is that when creating HTML files or looking at search results, we want a well defined order. The internal order does not matter all the much, for now.
